test/test_dispatcher.py

```diff
@@ -0,0 +1,448 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2015: Alignak team, see AUTHORS.txt file for contributors
+#
+# This file is part of Alignak.
+#
+# Alignak is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Alignak is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Alignak.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+This file test the dispatcher (distribute configuration to satellites)
+"""
+
+import time
+import requests_mock
+from alignak_test import AlignakTest
+from alignak.misc.serialization import unserialize
+
+
+class TestDispatcher(AlignakTest):
+    """
+    This class test the dispatcher  (distribute configuration to satellites)
+    """
+
+    def test_simple(self):
+        """
+        Simple test
+
+        have one realm and:
+        * 1 scheduler
+        * 1 poller
+        * 1 receiver
+        * 1 reactionner
+        * 1 broker
+
+        :return: None
+        """
+        self.setup_with_file('cfg/cfg_dispatcher_simple.cfg')
+        self.assertEqual(1, len(self.arbiter.dispatcher.realms))
+        for realm in self.arbiter.dispatcher.realms:
+            self.assertEqual(1, len(realm.confs))
+            for cfg in realm.confs.values():
+                self.assertTrue(cfg.is_assigned)
+        self.assertEqual(1, len(self.arbiter.dispatcher.schedulers))
+        self.assertEqual(4, len(self.arbiter.dispatcher.satellites))
+        for satellite in self.arbiter.dispatcher.satellites:
+            self.assertNotEqual({}, satellite.cfg['schedulers'], satellite.get_name())
+            self.assertEqual(1, len(satellite.cfg['schedulers']), 'must have 1 scheduler')
+
+        # check if scheduler has right the 6 hosts
+        self.assertEqual(6, len(self.schedulers['scheduler-master'].sched.hosts))
+
+    def test_simple_multi_schedulers(self):
+        """
+        Simple test (one realm) but with multiple schedulers:
+        * 2 scheduler
+        * 1 poller
+        * 1 receiver
+        * 1 reactionner
+        * 1 broker
+
+        :return: None
+        """
+        self.setup_with_file('cfg/cfg_dispatcher_simple_multi_schedulers.cfg')
+        self.assertEqual(1, len(self.arbiter.dispatcher.realms))
+        for realm in self.arbiter.dispatcher.realms:
+            self.assertEqual(2, len(realm.confs))
```

Comment:
Hum if you have 2 realm then you shuold have 2 broker. As far as I remember a realm is at least 1 sched and 1 brok. Unless this is a sub realm maybe?